### PR TITLE
Minor Swift 3.0 changes

### PR DESCRIPTION
**Pull request by Kacper20**
Cleanups in preparation for release 3.0 version.
Added one method that is new to iOS 9.0 & OSX 10.12 on Peripheral instance.

### CODE REVIEW

File: Source/BluetoothManager.swift [hunk 1]
```diff
@@ -128,7 +128,7 @@ public class BluetoothManager {
     public func scanForPeripherals(withServices serviceUUIDs: [CBUUID]?, options: [String: Any]? = nil)
         -> Observable<ScannedPeripheral> {
 
-            return Observable.deferred {
+            return.deferred {
```
Comment (Cierpliwy) [inline Source/BluetoothManager.swift]: Insert spaces after return keyword. Check it everywhere.

File: Source/Peripheral+Convenience.swift [hunk 2]
```diff
@@ -154,11 +148,9 @@ extension Peripheral {
      - Returns: Observable which emits `Next` with given characteristic when value is ready to read. Immediately after that
      `.Complete` is emitted.
      */
-    public func readValueForCharacteristic(with identifier: CharacteristicIdentifier) -> Observable<Characteristic> {
+    public func readValue(for identifier: CharacteristicIdentifier) -> Observable<Characteristic> {
         return characteristic(with: identifier)
-            .flatMap { characteristic in
-                return self.readValue(for: characteristic)
-            }
+            .flatMap { characteristic in return self.readValue(for: characteristic) }
```
Comment (Cierpliwy) [inline Source/Peripheral+Convenience.swift]: Use `$0` here?

File: Source/Peripheral+Convenience.swift [hunk 3]
```diff
@@ -56,16 +56,16 @@ extension Peripheral {
...
                 .flatMap { service -> Observable<Characteristic> in
                     if let characteristics = service.characteristics, let characteristic = characteristics.first(where: {
                         $0.uuid == identifier.uuid
                     }) {
-                        return Observable.just(characteristic)
+                        return.just(characteristic)
                     } else {
                         return service.discoverCharacteristics([identifier.uuid])
-                        .flatMap { Observable.from($0)}
+                            .flatMap { Observable.from($0) }
```
Comment (literator) [inline Source/Peripheral+Convenience.swift]: You could even write it like:

```swift
.flatMap(Observable.from)
```

File: Source/Peripheral+Convenience.swift [hunk 4]
```diff
@@ -80,15 +80,15 @@ extension Peripheral {
...
             return self.characteristic(with: identifier.characteristic)
                 .flatMap { characteristic -> Observable<Descriptor> in
                     if let descriptors = characteristic.descriptors,
                         let descriptor = descriptors.first(where: { $0.uuid == identifier.uuid }) {
-                        return Observable.just(descriptor)
+                        return.just(descriptor)
                     } else {
                         return characteristic.discoverDescriptors()
-                            .flatMap {Observable.from($0)}
+                            .flatMap { Observable.from($0) }
```
Comment (literator) [inline Source/Peripheral+Convenience.swift]: Maybe

```swift
.flatMap(Observable.from)
```

What do you think?

File: Source/Peripheral+Convenience.swift [hunk 5]
```diff
@@ -140,11 +136,9 @@ extension Peripheral {
      - Returns: Observable that emits `Next` with `Characteristic` instance every time when value has changed.
      It's **infinite** stream, so `.Complete` is never called.
      */
-    public func monitorValueUpdateForCharacteristic(with identifier: CharacteristicIdentifier) -> Observable<Characteristic> {
+    public func monitorValueUpdate(for identifier: CharacteristicIdentifier) -> Observable<Characteristic> {
```
Comment (literator) [inline Source/Peripheral+Convenience.swift]: Cool :+1:

File: Source/Peripheral+Convenience.swift (hunk 2 above)
Comment (literator) [inline Source/Peripheral+Convenience.swift]: Yeah,  you could do this then:

```swift
.flatMap(readValue)
```

Super short.

File: Source/Peripheral+Convenience.swift [hunk 6]
```diff
@@ -186,12 +176,10 @@ extension Peripheral {
...
      This is **infinite** stream of values.
      */
-    public func setNotificationAndMonitorUpdatesForCharacteristic(with identifier: CharacteristicIdentifier)
+    public func setNotificationAndMonitorUpdates(for identifier: CharacteristicIdentifier)
         -> Observable<Characteristic> {
             return characteristic(with: identifier)
-                .flatMap { char in
-                    return self.setNotificationAndMonitorUpdates(for: char)
-                }
+                .flatMap { return self.setNotificationAndMonitorUpdates(for: $0) }
```
Comment (literator) [inline Source/Peripheral+Convenience.swift]: Maybe:

```swift
.flatMap(setNotificationAndMonitorUpdates)
```

What do you think?

File: Source/Peripheral+Convenience.swift [hunk 7]
```diff
@@ -201,10 +189,10 @@ extension Peripheral {
      - Returns: Observable that emits `Next` with array of `Descriptor` instances, once they're discovered.
      Immediately after that `.Complete` is emitted.
      */
-    public func discoverDescriptorsForCharacteristic(with identifier: CharacteristicIdentifier) ->
+    public func discoverDescriptors(for identifier: CharacteristicIdentifier) ->
         Observable<[Descriptor]> {
         return characteristic(with: identifier)
-            .flatMap {self.discoverDescriptors(for: $0)}
+            .flatMap { self.discoverDescriptors(for: $0) }
```
Comment (literator) [inline Source/Peripheral+Convenience.swift]: Maybe:

```swift
.flatMap(discoverDescriptors)
```

What do you think?

File: Source/Peripheral+Convenience.swift [hunk 8]
```diff
@@ -213,11 +201,9 @@ extension Peripheral {
      - Returns: Observable that emits `Next` with `Descriptor` instance every time when write has happened.
      It's **infinite** stream, so `.Complete` is never called.
      */
-    public func monitorWriteForDescriptor(with identifier: DescriptorIdentifier) -> Observable<Descriptor> {
+    public func monitorWrite(for identifier: DescriptorIdentifier) -> Observable<Descriptor> {
         return descriptor(with: identifier)
-            .flatMap { desc in
-                return self.monitorWrite(for: desc)
-            }
+            .flatMap { return self.monitorWrite(for: $0) }
```
Comment (literator) [inline Source/Peripheral+Convenience.swift]: Maybe:

```swift
.flatMap(monitorWrite)
```

What do you think?

File: Source/Peripheral+Convenience.swift [hunk 9]
```diff
@@ -241,11 +225,9 @@ extension Peripheral {
      - Returns: Observable that emits `Next` with `Descriptor` instance every time when value has changed.
      It's **infinite** stream, so `.Complete` is never called.
      */
-    public func monitorValueUpdateForDescriptor(with identifier: DescriptorIdentifier) -> Observable<Descriptor> {
+    public func monitorValueUpdate(for identifier: DescriptorIdentifier) -> Observable<Descriptor> {
         return descriptor(with: identifier)
-            .flatMap { desc in
-                return self.monitorValueUpdate(for: desc)
-            }
+            .flatMap { return self.monitorValueUpdate(for: $0) }
```
Comment (literator) [inline Source/Peripheral+Convenience.swift]: Maybe:

```swift
.flatMap(monitorValueUpdate)
```

What do you think?

File: Source/Peripheral+Convenience.swift [hunk 10]
```diff
@@ -255,10 +237,8 @@ extension Peripheral {
      - Returns: Observable which emits `Next` with given descriptor when value is ready to read. Immediately after that
      `.Complete` is emitted.
      */
-    public func readValueForDescriptor(with identifier: DescriptorIdentifier) -> Observable<Descriptor> {
+    public func readValue(for identifier: DescriptorIdentifier) -> Observable<Descriptor> {
         return descriptor(with: identifier)
-            .flatMap { desc in
-                return self.readValue(for: desc)
-            }
+            .flatMap { return self.readValue(for: $0) }
```
Comment (literator) [inline Source/Peripheral+Convenience.swift]: Maybe:

```swift
.flatMap(readValue)
```

What do you think?

File: Source/Peripheral+Convenience.swift (hunk 4 above)
Comment (Kacper20) [inline Source/Peripheral+Convenience.swift]: This doesn't compile unfortunately.

File: Source/Peripheral+Convenience.swift (hunk 3 above)
Comment (Kacper20) [inline Source/Peripheral+Convenience.swift]: This doesn't compile unfortunately.

**Pull request merged**